Added new alert types, refs #10321

mcantelon opened the pull request:
Added ability to create "info" and "success" Bootstrap alerts.

This isn't needed for a feature, but I thought I needed it so I added it.

mcantelon committed:
Added new alert types, refs #10321

Added ability to create "info" and "success" Bootstrap alerts. Replaced
CSS ID for each notice type with an "app-alert" class (and tweaks JavaScript
to use it).

sevein commented on apps/qubit/templates/_alerts.php:
```diff
@@ -1,13 +1,8 @@
-    <?php echo $sf_user->getFlash('error', ESC_RAW) ?>
-  </div>
-<?php endif; ?>
+<?php foreach(array('notice', 'info', 'error', 'success') as $alertType): ?>
```
Nitpick... foreach is a statement, should we write as `foreach (...)` instead of `foreach(...)`?

mcantelon replied:
Ah right... thanks!

sevein commented on apps/qubit/templates/_alerts.php:
```diff
@@ -1,13 +1,8 @@
+      <?php echo $sf_user->getFlash($alertType, ESC_RAW) ?>
+    </div>
+  <?php endif; ?>
+<?php endforeach; ?>
```
Can you make sure that there is some space between the alert box and the header bar? IIRC there was no space at all between them making it visually not very pleasant.

mcantelon replied:
Good idea... I'll tweak the CSS.

sevein reviewed (commented):
Good idea, just a couple of suggestions!

mcantelon committed:
Add top margin to Bootstrap alerts, refs #10321

Added space above Bootstrap alerts so there's space between them and the
header.

mcantelon commented:
I've fixed those two issues. Thanks for ze review and let me know if there are any others.

sevein commented on plugins/arDominionPlugin/css/less/bootstrap-mod.less:
```diff
@@ -27,3 +27,8 @@
+
+// Upper padding for alerts
+.alert {
+  margin-top: 1em
```
One issue with this approach is that it's affecting all the cases where `.alert` is used, see `git grep -E "class=\".*alert.*\""`. Should you only target `#notice-alerts` instead? Additionally, not entirely sure but is it worth trying to find a better `id` for this element? e.g. `#app-alert` or `#global-alert` or similar?

mcantelon replied:
Good call... I'll make it more specific (and another change I should make is probably changing "#notice-alerts" to "#<notification type>-alert" so the ID only gets used once).

mcantelon replied:
(I ended up just getting rid of the ID and using app-alert as that's probably sufficient granularity for our needs.)

mcantelon commented:
OK, ready for another review. Thanks!

sevein commented:
👍

mcantelon commented:
Merged.
